src/main/java/unibook/logic/commands/EditCommand.java

```diff
@@ -57,6 +89,113 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editModuleDescriptor = null;
+    }
+
+    /**
+     * @param index of the module in the filtered module list to edit
+     * @param editModuleDescriptor details to edit the module with
+     */
+    public EditCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = index;
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+        this.editPersonDescriptor = null;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException, ModuleNotFoundException {
+        requireNonNull(model);
+
+        if (this.editModuleDescriptor == null) {
+            List<Person> lastShownList = model.getFilteredPersonList();
+            List<Module> latestModList = model.getFilteredModuleList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            Person personToEdit = lastShownList.get(index.getZeroBased());
+            Module checkMod = null;
+            if (!editPersonDescriptor.getModules().equals(Optional.empty()) && latestModList.size() != 0) {
+                checkMod = editPersonDescriptor.getModules().get().iterator().next();
+                if (!latestModList.contains(checkMod)) {
+                    throw new ModuleNotFoundException();
+                }
+            }
+
+            Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+
+            // Temporarily using tags to identify whether person is prof or student
+            // Need to change the EditPersonDescriptor to include type of person, maybe EditProfDescriptor etc
+            String editedPersonType = null;
```

Comment:
Don't leave it like this, if set to this edit function will not work. Remember that v1.2 needs to be functional, additional features can be spared but the base functionality of each command **must** be there.

Consider simply searching for this person you are trying to edit in the list of students and list of profs in the module regardless of the true subtype that person is supposed to be, since a person can only be a student or a prof, there will only be one match in students/profs, just change that one match.